test/unit/ConfigurationTest.php

```diff
@@ -306,4 +313,60 @@ public function getValidatorShouldReturnTheConfiguredValidator(): void
 
         self::assertSame($this->validator, $config->getValidator());
     }
+
+    /**
+     * @test
+     *
+     * @covers \Lcobucci\JWT\Configuration::getValidationConstraints
+     *
+     * @uses \Lcobucci\JWT\Configuration::forUnsecuredSigner
+     * @uses \Lcobucci\JWT\Configuration::__construct
+     * @uses \Lcobucci\JWT\Signer\None
+     * @uses \Lcobucci\JWT\Signer\Key
+     */
+    public function getValidationConstraintsShouldReturnTheDefaultWhenItWasNotConfigured(): void
+    {
+        $config                = Configuration::forUnsecuredSigner();
+        $validationConstraints = $config->getValidationConstraints();
+
+        self::assertSame([], $validationConstraints);
+        self::assertNotSame($this->validationConstraints, $validationConstraints);
+    }
+
+    /**
+     * @test
+     *
+     * @covers \Lcobucci\JWT\Configuration::getValidationConstraints
+     * @covers \Lcobucci\JWT\Configuration::setValidationConstraints
+     *
+     * @uses \Lcobucci\JWT\Configuration::forUnsecuredSigner
+     * @uses \Lcobucci\JWT\Configuration::__construct
+     * @uses \Lcobucci\JWT\Signer\None
+     * @uses \Lcobucci\JWT\Signer\Key
+     */
+    public function getValidationConstraintsShouldReturnTheConfiguredValidator(): void
+    {
+        $config = Configuration::forUnsecuredSigner();
+        $config->setValidationConstraints($this->validationConstraints);
+
+        self::assertSame($this->validationConstraints, $config->getValidationConstraints());
+    }
+
+    /**
+     * @test
+     *
+     * @covers \Lcobucci\JWT\Configuration::setValidationConstraints
+     *
+     * @uses \Lcobucci\JWT\Configuration::forUnsecuredSigner
+     * @uses \Lcobucci\JWT\Configuration::__construct
+     * @uses \Lcobucci\JWT\Signer\None
+     * @uses \Lcobucci\JWT\Signer\Key
+     */
+    public function setValidationConstraintsShouldThrowExceptionWhenConstraintsArrayIsNotValid(): void
+    {
+        self::expectException(\InvalidArgumentException::class);
```

Comment:
This would not be an `InvalidArgumentException` but rather a type error if we use the variadic form.